transaction.py

```diff
@@ -41,9 +62,59 @@ def __bytes__(self):
             pack_g1(self.g_H)
         )
 
+class GrothProof(object):
+    def __init__(self, rand):
+        self.g_A = rand.b(48)
+        self.g_B = rand.b(96)
+        self.g_C = rand.b(48)
 
-class JoinSplit(object):
+    def __bytes__(self):
+        return (
+            self.g_A +
+            self.g_B +
+            self.g_C
+        )
+
+class SpendDescription(object):
+    def __init__(self, rand):
+        self.cv = find_group_hash(b'TVRandPt', rand.b(32))
+        self.anchor = Fq(leos2ip(rand.b(32)))
+        self.nullifier = rand.b(32)
+        self.rk = rand.b(32)
+        self.proof = GrothProof(rand)
+        self.spendAuthSig = rand.b(64) # Invalid
```

Comment:
what does the invalid comment mean exactly?

Reply:
It's just indicating that it's invalid (a random blob) because the purpose is to test the serialization and SignatureHash behavior, not components like signatures or proofs.

Reply:
right but that seems true of all the fields, so I didn't understand what was special about this line.

Reply:
You make a fair point that all the data is test data.  Its just a human friendly reminder that a computation or derivation has not taken place, that the field is junk.  To be consistent with existing fields labelled this way, this PR sticks with signature and proof.